app/services/product_drive_destroy_service.rb

```diff
@@ -0,0 +1,64 @@
+class ProductDriveDestroyService
+  attr_reader :product_drive, :user, :organization
+
+  def initialize(product_drive, user, organization)
+    @product_drive = product_drive
+    @user = user
+    @organization = organization
+  end
+
+  def self.call(product_drive, user, organization)
+    new(product_drive, user, organization).call
```

Comment:
um... my point was that we do not need to call `new` at all. All methods can be class-level methods (you can wrap the whole shebang in a `class << self` block to make it simpler).